src/posix.mak

```diff
@@ -99,7 +100,7 @@ else
   HOST_DMD_RUN=$(HOST_DMD) -conf=$(dir $(HOST_DMD))dmd.conf
 endif
 
-RUN_BUILD = $(GENERATED)/build HOST_DMD="$(HOST_DMD)" CXX="$(HOST_CXX)" OS=$(OS) BUILD=$(BUILD) MODEL=$(MODEL) AUTO_BOOTSTRAP="$(AUTO_BOOTSTRAP)" DOCDIR="$(DOCDIR)" STDDOC="$(STDDOC)" DOC_OUTPUT_DIR="$(DOC_OUTPUT_DIR)" MAKE="$(MAKE)" --called-from-make
+RUN_BUILD = $(GENERATED)/build --called-from-make OS="$(OS)" BUILD="$(BUILD)" MODEL="$(MODEL)" HOST_DMD="$(HOST_DMD)" CXX="$(HOST_CXX)" AUTO_BOOTSTRAP="$(AUTO_BOOTSTRAP)" DOCDIR="$(DOCDIR)" STDDOC="$(STDDOC)" DOC_OUTPUT_DIR="$(DOC_OUTPUT_DIR)" MAKE="$(MAKE)" VERBOSE="$(VERBOSE)" ENABLE_RELEASE="$(ENABLE_RELEASE)" ENABLE_DEBUG="$(ENABLE_DEBUG)" ENABLE_ASSERTS="$(ENABLE_ASSERTS)" ENABLE_UNITTEST="$(ENABLE_UNITTEST)" ENABLE_PROFILE="$(ENABLE_PROFILE)" ENABLE_COVERAGE="$(ENABLE_COVERAGE)" DFLAGS="$(DFLAGS)"
```

Comment:
IIRC `build.d` parses `ENABLE_` from both CLI arguments and environment arguments and as GNUMake forwards all environment variables, this shouldn't be necessary?
Or was this done for explicitness?

Reply:
GNU make apparently automatically exports all variables set in the cmdline to the environment, but DM make doesn't (at least not on Windows). I wasn't sure whether DM make is supposed to be supported on Posix systems.

Reply:
Well, posix.mak can't even be called with DM make as it uses many features that DM make doesn't support. It will outright fail to even parse the Makefile.

As we are about to finally deprecate these mak files, I'm fine either way. I was just curious, because I thought it wasn't technically necessary.

CC @MoonlightSentinel: you were the one who added the make->build.d call initially. Do you have any objections for or against this?

Reply:
This line was actually introduced by @marler8997 in e07dc9539abd3654c2fdf9996b177fb363603878.

Anyway, i don't think there is any harm in adding those variables because it doesn't change the behaviour and we're phasing out the makefiles anyway.

Reply:
Hmm, I thought dm make did add variables to its environment, which is then available to any invoked command. I wrote that program 35 years ago, I guess I'd have to go look at the source code :-)